src/main/java/com/gmail/woosay333/onlinebookstore/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,29 @@
+package com.gmail.woosay333.onlinebookstore.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String firstFieldName;
+    private String secondFieldName;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.firstFieldName = constraintAnnotation.first();
+        this.secondFieldName = constraintAnnotation.second();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        try {
+            Object firstValue = new BeanWrapperImpl(value).getPropertyValue(firstFieldName);
+            Object secondValue = new BeanWrapperImpl(value).getPropertyValue(secondFieldName);
+
+            return firstValue == null && secondValue == null
+                    || firstValue != null && firstValue.equals(secondValue);
+        } catch (Exception ex) {
+            return false;
```

Comment:
I'd rather throw ex instead of retuning false because it would be our mistake as a programmer if we put non existing field as a parameter to the annotation